### ISSUE TITLE
Plugin keyboard shortcuts remain even after uninstalling the plugin

### ISSUE THREAD

**Issue by personalizedrefrigerator** [laurent22/joplin#10033]
### Operating system

Linux

### Joplin version

2.14.17

### Desktop version info

Joplin 2.14.17 (dev, linux)

Client ID: 44da07c87ebf4662aa0e0f97f77f7f38
Sync Version: 3
Profile Version: 46
Keychain Supported: No

Revision: 99e36ee92 (pr/cm/upgrade-cm6)

Backup: 1.4.0
CodeMirror 6 snippets: 0.0.5

### Current behaviour

> [!NOTE]
>
> Original issue report: https://discourse.joplinapp.org/t/shortcut-assignment-issue-for-add-on/36274
>

Currently, after uninstalling a plugin, commands from that plugin that have been associated with a keyboard shortcut show up as "invalid" in the keyboard settings pane:

![screenshot: Freehand drawing shortcut shows up as invalid](https://github.com/laurent22/joplin/assets/46334387/29f81812-ab4d-4eed-bbd4-29373cdef2c4)

Additionally, these shortcuts cannot be reassigned until after clearing them from the original command.

### Expected behaviour

A few things that may make more sense:
1. The command could appear with its original name (rather than the plugin ID) and include `(uninstalled)` rather than `(invalid)`
2. Such shortcuts could be hidden in the search results, attempting to assign these shortcuts to a new plugin would then clear the original shortcut and re-assign.

**Comment by laurent22** [comment]
> Additionally, these shortcuts cannot be reassigned until after clearing them from the original command.

I think this was the issue mentioned in the forum thread? So the user would need to clear the shortcut, which I feel is acceptable? I don't quite understand how the changes you suggest would help with this issue

**Comment by personalizedrefrigerator** [comment]
> I think this was the issue mentioned in the forum thread?

Yes, I moved it here because it seemed like confusing behavior from a user's perspective.

>  I don't quite understand how the changes you suggest would help with this issue.

My goal with the items under "expected behavior" was to make the invalid commands easier to discover and/or easier to deassociate a keyboard shortcut from. On reflection, I don't think option 1 above would do that. Option 2 would be different from the current workflow for reassigning other keyboard shortcuts.

Closing.
